docs/source/howto/plugin_codes.rst

```diff
@@ -4,172 +4,188 @@
+We define the ``retrieve_list`` of filenames that the engine should retrieve from the directory where the job ran after it has finished.
+The engine will store these files in a |FolderData| node that will be attached as an output node to the calculation with the label ``retrieved``.
 There are :ref:`other file lists available<topics:calculations:usage:calcjobs:file_lists>` that allow you to easily customize how to move files to and from the remote working directory in order to prevent the creation of unnecessary copies.
 
-This was a minimal example of how to implement the |CalcJob| class to interface AiiDA with an external code.
-For more detailed information and advanced functionality on the |CalcJob| class, refer to the Topics section on :ref:`defining calculations <topics:calculations:usage>`.
+This was an example of how to implement the |CalcJob| class to interface AiiDA with an external code.
+For more details on the |CalcJob| class, refer to the Topics section on :ref:`defining calculations <topics:calculations:usage>`.
 
 .. _how-to:plugin-codes:parsing:
 
 Parsing the outputs
 ===================
 
-The parsing of the output files generated by a |CalcJob| is optional and can be used to store (part of) their information as AiiDA nodes, which makes the data queryable and therefore easier to access and analyze.
-To enable |CalcJob| output file parsing, one should subclass the |Parser| class and implement the :py:meth:`~aiida.parsers.parser.Parser.parse` method.
-The following is an example implementation, as shipped with ``aiida-core``, to parse the outputs of the :py:class:`~aiida.calculations.arithmetic.add.ArithmeticAddCalculation` discussed in the previous section:
+Parsing the output files produced by your code into AiiDA nodes is optional, but it can make your data queryable and therefore easier to access and analyze.
+
+We start our ``parser.py`` plugin by subclassing the |Parser| class and implementing its :py:meth:`~aiida.parsers.parser.Parser.parse` method.
 
 .. literalinclude:: ../../../aiida/parsers/plugins/arithmetic/add.py
     :language: python
-    :pyobject: ArithmeticAddParser
+    :pyobject: SimpleArithmeticAddParser
+
+Before the ``parse()`` method is called, two important attributes are set on the |Parser|  instance:
+
+  1. ``self.retrieved``: An instance of |FolderData|, which points to the folder containing all output files produced by the code and provides the means to :py:meth:`~aiida.orm.nodes.node.Node.open` any file it contains.
+
+  2. ``self.node``: The :py:class:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode` representing the finished calculation, which e.g. provides access to all of its inputs (``self.node.inputs``).
+
+We start by opening the output file, whose filename we get from the ``self.node`` attribute via the :py:meth:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode.get_option` convenience method.
```

Comment:
```suggestion
The output of the code is obtained by reading the content of the output file, whose filename we get from the ``self.node`` attribute via the :py:meth:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode.get_option` convenience method.
The read content is cast to an integer, as it should contain the sum that was produced by the ``bash`` code.
```

Reply:
```suggestion
The :py:meth:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode.get_option` convenience method is used to get the filename of the output file.
Its content is cast to an integer, since the output file should contain the sum produced by the ``aiida.in`` bash script.
```